anta/reporter/__init__.py

```diff
@@ -211,3 +215,41 @@ def report_summary_hosts(
                     str(list_failure),
                 )
         return table
+
+
+class ReportJinja:
+    """Report builder based on a Jinja2 template."""
+
+    def __init__(self, template_path: str) -> None:
+        if os.path.isfile(template_path):
+            self.tempalte_path = template_path
+        else:
+            raise FileNotFoundError(f"template file is not found: {template_path}")
+
+    def render(self, data: List[Dict[str, Any]], trim_blocks: bool = True, lstrip_blocks: bool = True) -> str:
+        """
+        Build a report based on a Jinja2 template
+
+        Report is built based on a J2 template provided by user.
+        Data structure sent to template is:
+        data = [
+            {
+                name: ...,
+                test: ...,
+                result: ...,
+                messages: [...]
+            }
+        ]
+
+        Args:
+            data (List[Dict[str, Any]]): List of results from ResultManager.get_results
```

Comment:
`get_results` has to be in JSON format right? Maybe add it in the description: `ResultManager.get_results(output_format="json")`